cirque/common/cirqueprompt.py

```diff
@@ -0,0 +1,340 @@
+        ret = homes[args.home_id].get_home_devices()
+        print("home id: {}".format(args.home_id))
+        print("home devices: {}".format(pp.pformat(ret)))
+
+    def help_homes(self):
+        print(
+            textwrap.dedent("""\
+                list all the homes
+
+                Usage:
+                    homes
+                Return:
+                    home_id (list)
+            """))
+
+    def do_homes(self, inp):
+        print(list(homes.keys()))
+
+    def help_run_exec(self):
+        print(
+            textwrap.dedent("""\
+                run command in a specific docker container(device id)
+
+                Args:
+                    --home_id (str): home_id. if not given, automatically
+                            set to current home id.
+                    --device_id (str): device id
+                    --command (str): if command includes --(switch) repaces to
+                            __(under line), the api will replace them back
+                            automatically.
+                    --stream (bool): default False, if switch is set, means
+                            True
+                Usage:
+                    cirque> run_exec [--home_id <home_id>] --device_id
+                            <device_id> --command <command> [--stream]
+                Returns:
+                    command exectue result (str)
+            """))
+
+    def do_run_exec(self, inp):
+        parser = argparse.ArgumentParser()
+        parser.add_argument('--home_id', type=str, default=None)
+        parser.add_argument('--device_id', type=str, required=True)
+        parser.add_argument('--command', type=str, nargs='+', required=True)
+        parser.add_argument('--stream', action='store_true', default=False)
+        args = parser.parse_args(inp.split())
+        if not args.home_id:
+            args.home_id = str(self.home_id - 1)
+        else:
+            if not self._valid_home_id(args.home_id):
+                print("home id: {} is not existing!!".format(args.home_id))
+                return
+        args.command = " ".join(args.command)
+        args.command = args.command.replace('__', '--')
+        print("args.command: {}".format(args.command))
+        ret = homes[args.home_id].execute_device_cmd(args.command,
+                                                     args.device_id,
+                                                     args.stream)
+        print("command result:\n{}".format(ret))
+
+    def help_version(self):
+        print(
+            textwrap.dedent("""\
+                displaycurrent cirque version.
+
+                Usage:
+                    version
+                Returne:
+                    version (str)
+            """))
+
+    def do_version(self, inp):
+        print(self.version)
+
+    def emptyline(self):
+        pass
+
+    def _valid_home_id(self, home_id):
+        return home_id in homes
+
+
+if __name__ == '__main__':
+    import sys
+    sys.exit(
```

Comment:
Lint: Is it a must to have a new line here?

Reply:
ack!